File: csharp/Betfair.ESAClient/Betfair.ESAClient/Cache/OrderMarketSnap.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Betfair.ESAClient.Cache {
    /// <summary>
    /// Thread safe atomic snapshot of a market.
    /// Reference only changes if the snapshot changes:
    /// i.e. if snap1 == snap2 then they are the same.
    /// (same is true for sub-objects)
    /// </summary>
    public class OrderMarketSnap {
        public string MarketId { get; internal set; }
        public bool IsClosed { get; internal set; }
        public IEnumerable<OrderMarketRunnerSnap> OrderMarketRunners { get; internal set; }

        public override string ToString() {
            return "OrderMarketSnap{" + "MarketId=" + MarketId + ", IsClosed=" + IsClosed + ", OrderMarketRunners=" + String.Join(", ", OrderMarketRunners) + "}";
        }
    }
}

File: java/client/src/main/java/com/betfair/esa/client/cache/util/OrderMarketSnap.java
package com.betfair.esa.client.cache.util;

public class OrderMarketSnap {

    private String marketId;
    private boolean isClosed;
    private Iterable<OrderMarketRunnerSnap> orderMarketRunners;

    public boolean isClosed() {
        return isClosed;
    }

    public void setClosed(boolean closed) {
        isClosed = closed;
    }

    public Iterable<OrderMarketRunnerSnap> getOrderMarketRunners() {
        return orderMarketRunners;
    }

    public void setOrderMarketRunners(Iterable<OrderMarketRunnerSnap> orderMarketRunners) {
        this.orderMarketRunners = orderMarketRunners;
    }

    public String getMarketId() {
        return marketId;
    }

    public void setMarketId(String marketId) {
        this.marketId = marketId;
    }

    @Override
    public String toString() {
        return "OrderMarketSnap{"
                + "marketId='"
                + marketId
                + '\''
                + ", isClosed="
                + isClosed
                + ", orderMarketRunners="
                + orderMarketRunners
                + '}';
    }
}
